### PR TITLE
Allow more default sink parameters

### PR DESCRIPTION
**Pull request by Lokdei**
Hi

This PR allows to add sink parameters 
- restrictedToMinimumLevel
- outputTemplate
- formatProvider 
- levelSwitch

And 1 test to check if the restrictedToMinimumLevel works as expected.

Would you please give feedback upon this and consider merging it ?

The tests all pass.

### CODE REVIEW

**Comment by sandermvanvliet** [comment]
Hi @lokdei thanks for your PR!
I’ll give this a look over the weekend

File: src/Serilog.Sinks.InMemory/InMemorySink.cs [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 using Serilog.Core;
 using Serilog.Events;
+using Serilog.Formatting;
 
 namespace Serilog.Sinks.InMemory
 {
     public class InMemorySink : ILogEventSink, IDisposable
     {
         private static readonly AsyncLocal<InMemorySink> LocalInstance = new AsyncLocal<InMemorySink>();
+        private readonly ITextFormatter _formatProvider;
```
Comment (sandermvanvliet) [inline src/Serilog.Sinks.InMemory/InMemorySink.cs:13]: Given that the `_formatProvider` is never used I would recommend removing it, that would also remove the necessity for `GetInstance()` as you can use `Instance` like before.

File: src/Serilog.Sinks.InMemory/InMemorySinkExtensions.cs [hunk 2]
```diff
@@ -1,13 +1,65 @@
...
+        public static LoggerConfiguration InMemory(
+            this LoggerSinkConfiguration sinkConfiguration,
+            ITextFormatter formatter,
+            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum,
+            LoggingLevelSwitch levelSwitch = null)
+        {
+            if (sinkConfiguration == null) throw new ArgumentNullException(nameof(sinkConfiguration));
+            if (formatter == null) throw new ArgumentNullException(nameof(formatter));
+
+            return sinkConfiguration.Sink(InMemorySink.GetInstance(formatter), restrictedToMinimumLevel, levelSwitch);
```
Comment (sandermvanvliet) [inline src/Serilog.Sinks.InMemory/InMemorySinkExtensions.cs:62]: As mentioned in the other comment, I don't think you need to supply the `formatter` here as it isn't used by the sink.

**Comment by Lokdei** [comment]
Hey

I added a section in the Readme: "Creating a logger" with setup parameter examples.

I also removed the unused `formatter` from the constructor and GetInstance() call.

**Comment by sandermvanvliet** [comment]
@Lokdei  version [0.5.0](https://www.nuget.org/packages/Serilog.Sinks.InMemory/0.5.0) has been pushed to NuGet and should be visible shortly

**Pull request merged**